pkg/agent/supportbundlecollection/support_bundle_controller_test.go

```diff
@@ -0,0 +1,266 @@
+		},
+		{
+			name:                    "Add SupportBundleCollection without url prefix",
+			supportBundleCollection: generateSupportbundleCollection("supportBundle2", "10.220.175.92:22/root/supportbundle"),
+		},
+	}
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			newAgentDumper = func(fs afero.Fs, executor exec.Interface, ovsCtlClient ovsctl.OVSCtlClient, aq agentquerier.AgentQuerier, npq querier.AgentNetworkPolicyInfoQuerier, since string, v4Enabled, v6Enabled bool) support.AgentDumper {
+				return &mockAgentDumper{}
+			}
+			defer func() {
+				newAgentDumper = support.NewAgentDumper
+			}()
+			controller, clientset := newFakeController(t)
+			var bundleStatus *cpv1b2.SupportBundleCollectionStatus
+			clientset.AddReactor("update", "supportbundlecollections/status", k8stesting.ReactionFunc(func(action k8stesting.Action) (bool, runtime.Object, error) {
+				bundleStatus = action.(k8stesting.UpdateAction).GetObject().(*cpv1b2.SupportBundleCollectionStatus)
+
+				return false, bundleStatus, nil
+			}))
+			controller.addSupportBundleCollection(tt.supportBundleCollection)
+			assert.NoError(t, controller.syncSupportBundleCollection(tt.supportBundleCollection.Name))
+			assert.Equal(t, true, bundleStatus.Nodes[0].Completed)
+		})
+	}
+}
+
+func TestSupportBundleCollectionDelete(t *testing.T) {
+	controller, _ := newFakeController(t)
+	deletedBundle := generateSupportbundleCollection("deletedBundle", "sftp://10.220.175.92/root/supportbundle")
+	controller.addSupportBundleCollection(deletedBundle)
+	controller.deleteSupportBundleCollection(deletedBundle)
+	assert.NoError(t, controller.syncSupportBundleCollection("deletedBundle"))
+}
+
+func TestSupportBundleCollectionRetry(t *testing.T) {
+	newAgentDumper = func(fs afero.Fs, executor exec.Interface, ovsCtlClient ovsctl.OVSCtlClient, aq agentquerier.AgentQuerier, npq querier.AgentNetworkPolicyInfoQuerier, since string, v4Enabled, v6Enabled bool) support.AgentDumper {
+		return &mockAgentDumper{}
+	}
+	defer func() {
+		newAgentDumper = support.NewAgentDumper
+	}()
+	bundle4 := generateSupportbundleCollection("supportBundle4", "sftp://10.220.175.92/root/supportbundle")
+	controller, clientset := newFakeController(t)
+	controller.sftpUploader = &testFailedUploader{}
+	var bundleStatus *cpv1b2.SupportBundleCollectionStatus
+	clientset.PrependReactor("update", "supportbundlecollections/status", k8stesting.ReactionFunc(func(action k8stesting.Action) (bool, runtime.Object, error) {
+		bundleStatus = action.(k8stesting.UpdateAction).GetObject().(*cpv1b2.SupportBundleCollectionStatus)
+
+		return false, bundleStatus, nil
+	}))
+	controller.addSupportBundleCollection(bundle4)
+	assert.Error(t, controller.syncSupportBundleCollection(bundle4.Name))
+	assert.Equal(t, false, bundleStatus.Nodes[0].Completed)
+}
+
+func TestSupportBundleCollectionDumpFailure(t *testing.T) {
```

Comment:
ditto, could be merged too, just need to parameterize agentDumper

Reply:
Done